addOns/importLogFiles/src/main/java/org/zaproxy/zap/extension/importLogFiles/ImportLogAPI.java

```diff
@@ -274,11 +274,14 @@ private static ApiResponseElement processLogs(
 
         if (!targetFile.isFile() /* && !FileAlreadyExists(new File(sourceFilePath)) */) {
             try {
-                targetFile.createNewFile();
+                boolean success = targetFile.createNewFile();
```

Comment:
Better do proper checks and not explicitly create the file (let it be created when writing).

Reply:
You mean just skip the call to `createNewFile`?

Reply:
Yes, that's unnecessary. It can just check if the file exists or not.

Reply:
How about this (vs. the original):

```diff
diff --git a/addOns/importLogFiles/src/main/java/org/zaproxy/zap/extension/importLogFiles/ImportLogAPI.java b/addOns/importLogFiles/src/main/java/org/zaproxy/zap/extension/importLogFiles/ImportLogAPI.java
index a62b5fe..e3abe90 100644
--- a/addOns/importLogFiles/src/main/java/org/zaproxy/zap/extension/importLogFiles/ImportLogAPI.java
+++ b/addOns/importLogFiles/src/main/java/org/zaproxy/zap/extension/importLogFiles/ImportLogAPI.java
@@ -263,34 +263,21 @@
             String httpPOSTData) {
         // Not appending the file with client state info as REST should produce a resource based on
         // the request indefinitely.
-        String sourceFilePath = filePath;
         String targetfileName =
-                sourceFilePath.substring(
-                                sourceFilePath.lastIndexOf("\\") + 1,
-                                sourceFilePath.lastIndexOf("."))
+                filePath.substring(
+                        filePath.lastIndexOf("\\") + 1,
+                        filePath.lastIndexOf("."))
                         + ".txt";
         String absoluteTargetFilePath = getLoggingStorageDirectory(logType) + "\\" + targetfileName;
         File targetFile = new File(absoluteTargetFilePath);
 
-        if (!targetFile.isFile() /* && !FileAlreadyExists(new File(sourceFilePath)) */) {
-            try {
-                targetFile.createNewFile();
-                // TODO investigate how to check for uniqueness of the file. Potentially hashing
-                // (md5) the string[] of the read
-                // file. Might be overkill?
-                // appendAddedFilesHashes(targetFile);
-            } catch (Exception ex) {
-                return new ApiResponseElement(
-                        "Parsing logs files to ZAPs site tree",
-                        "Failed - Could not create file on server");
-            }
-        } else {
+        if (targetFile.isFile()) {
             return new ApiResponseElement(
                     "Parsing logs files to ZAPs site tree", "Not processed - File already added");
         }
 
         if (httpPOSTData == null) {
-            try (BufferedReader br = new BufferedReader(new FileReader(sourceFilePath));
+            try (BufferedReader br = new BufferedReader(new FileReader(filePath));
                     BufferedWriter wr = new BufferedWriter(new FileWriter(targetFile))) {
                 String sCurrentLine;
 
```

Plus I guess at 290 and 298 it'll need to return a ApiException (and 275 should be an ApiException not ApiResponseElement?)?

Reply:
Yeah, it would make more sense, but leave as is for now? This is something that can be moved to exim add-on (and can be cleaned up further then).

Reply:
https://github.com/zaproxy/zaproxy/issues/6579